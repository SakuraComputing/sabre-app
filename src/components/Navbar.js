import React from 'react';

export default class Navbar extends React.Component {

    constructor(props) {
        super(props);
    }

    onHamburgerClick = () =>  {
      this.props.onHamburgerClick && this.props.onHamburgerClick();
    };

    render(){
        return (
            <nav className="navbar navbar-default navbar-fixed-top">
                <div className="container-fluid">
                    <div className="navbar-header">
                        <button type="button" className="navbar-btn navbar-toggle collapsed" onClick={this.onHamburgerClick}>
                            <span className="sr-only">Toggle Navigation</span>
                            <span className="icon-bar"></span>
                            <span className="icon-bar"></span>
                            <span className="icon-bar"></span>
                        </button>
                    </div>
                    {this.props.children}
                </div>
            </nav>
        );
    }
}