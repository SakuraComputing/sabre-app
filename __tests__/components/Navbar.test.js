import React from 'react';
import Navbar from '../../src/components/Navbar';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

describe('Navbar test',() => {

   let navbar;

   let mockCallBack = jest.fn();

   describe('#onHamburgerClick', () => {
      describe('no callback provided',() => {
         it('should not error', () => {
             navbar = shallow(<Navbar />);
             navbar.instance().onHamburgerClick();
         });
      });

      describe('when callback provided should invoke', () => {
         beforeEach(() => {
             navbar = shallow(<Navbar onHamburgerClick={mockCallBack}/>)
             navbar.instance().onHamburgerClick();
         });
         it('should render the hamburger menu',() => {
            expect(mockCallBack).toHaveBeenCalled();
         });
      });
   });

   it('should render the navbar',() => {
       // const wrapper = shallow(<Navbar />);
       // expect(wrapper).toMatchSnapshot();
       const wrapper = renderer.create(<Navbar />);
       let tree = wrapper.toJSON();
       expect(tree).toMatchSnapshot();
   })
});
