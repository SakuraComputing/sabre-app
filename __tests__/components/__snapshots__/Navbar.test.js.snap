// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar test should render the navbar 1`] = `
<nav
  className="navbar navbar-default navbar-fixed-top"
>
  <div
    className="container-fluid"
  >
    <div
      className="navbar-header"
    >
      <button
        className="navbar-btn navbar-toggle collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="sr-only"
        >
          Toggle Navigation
        </span>
        <span
          className="icon-bar"
        />
        <span
          className="icon-bar"
        />
        <span
          className="icon-bar"
        />
      </button>
    </div>
  </div>
</nav>
`;
